### PR TITLE
TINY-6920: Change event no longer triggered on first modification

### PR DESCRIPTION
**Pull request by HAFRMO**
Related Ticket:

Description of Changes:
The change event is not fired anymore, since the dirty event is fired instead.

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/` for new features (if applicable)

Review:
* [x] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):
Fixes https://github.com/tinymce/tinymce/issues/6366

### CODE REVIEW

**Review by spocke — APPROVED** [review]

File: modules/tinymce/src/core/test/ts/browser/EditorTest.ts [hunk 1]
```diff
@@ -45,6 +45,24 @@ describe('browser.tinymce.core.EditorTest', () => {
     editor.off('change');
   });
 
+  it('TINY-6920: Do not fire change event at first typed character', () => {
```
Comment (spocke) [inline modules/tinymce/src/core/test/ts/browser/EditorTest.ts:48]: Should this be in the UndoManagerTest unsure.

Comment (lnewson) [reply modules/tinymce/src/core/test/ts/browser/EditorTest.ts:48]: I'd agree this is more suited to an undo manager test, so it should be somewhere closer to there.

File: modules/tinymce/src/core/test/ts/browser/EditorTest.ts [hunk 2]
```diff
@@ -45,6 +45,24 @@ describe('browser.tinymce.core.EditorTest', () => {
...
+    const editor = hook.editor();
+    let changeEventCounter = 0;
+
+    const onChange = () => {
+      changeEventCounter++;
+    };
+
+    editor.setContent('');
+    editor.setDirty(false);
```
Comment (lnewson) [inline modules/tinymce/src/core/test/ts/browser/EditorTest.ts:57]:
```suggestion
    editor.resetContent('');
```
This is a better way to reset the content, dirty state and clear the undomanager state.

**Review by lnewson — APPROVED** [review]

**Comment by lnewson** [comment]
@HAFRMO also since this fixes a GitHub issue don't forget to add that to the description please.

File: modules/tinymce/src/core/test/ts/browser/EditorTest.ts (hunk 2 above)
Comment (HAFRMO) [reply modules/tinymce/src/core/test/ts/browser/EditorTest.ts]: Useful.

**Review by ltrouton — APPROVED** [review]

**Pull request merged**